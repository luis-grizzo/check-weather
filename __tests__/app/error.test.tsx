import { fireEvent, render, screen } from '@testing-library/react'

import Error from '@/app/error'

const pushMock = jest.fn()

jest.mock('next/navigation', () => ({
  useRouter() {
    return {
      push: pushMock
    }
  }
}))

describe('Error', () => {
  beforeEach(() => {
    render(<Error error={{ name: 'error name', message: 'error message' }} />)
  })

  it('should match snapshot', () => {
    const { container } = render(
      <Error error={{ name: 'error name', message: 'error message' }} />
    )

    expect(container).toMatchSnapshot()
  })

  it('should call router push on button click', () => {
    const eButton = screen.getByRole('button') as HTMLButtonElement

    fireEvent.click(eButton)

    expect(pushMock).toHaveBeenCalledWith('/')
  })
})
