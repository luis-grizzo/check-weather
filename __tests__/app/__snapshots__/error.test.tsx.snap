// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should match snapshot 1`] = `
<div>
  <main
    class="flex flex-col items-center justify-center gap-8 h-dvh container mx-auto p-8"
  >
    <img
      alt=""
      class="w-24 aspect-square"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
    <h2
      class="text-lg"
    >
      Something went wrong!
    </h2>
    <div
      class="flex items-start gap-4 p-4 text-sm text-red-950 bg-red-100/60 border-red-600/10 border-1 rounded-2xl backdrop-blur"
      role="dialog"
    >
      <img
        alt=""
        class="w-5 aspect-square"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      Please check your browser's location access permissions and refresh the page or try again.
    </div>
    <button
      class="flex items-center justify-center py-3 px-4 bg-neutral-100/60 border-1 border-neutral-600/10 rounded-lg"
      role="button"
    >
      Try again
    </button>
  </main>
</div>
`;
